internal/scheduler/scheduling/context/queue.go

```diff
@@ -197,6 +216,25 @@ func (qctx *QueueSchedulingContext) addJobSchedulingContext(jctx *JobSchedulingC
 	return evictedInThisRound, nil
 }
 
+func (qctx *QueueSchedulingContext) preemptJob(jctx *JobSchedulingContext) (bool, error) {
+	jobId := jctx.Job.Id()
+
+	pcName := jctx.Job.PriorityClassName()
+	rl := jctx.Job.AllResourceRequirements()
+	_, scheduledInThisRound := qctx.SuccessfulJobSchedulingContexts[jobId]
+	if scheduledInThisRound {
+		qctx.ScheduledResourcesByPriorityClass[pcName] = qctx.ScheduledResourcesByPriorityClass[pcName].Subtract(rl)
+		delete(qctx.SuccessfulJobSchedulingContexts, jobId)
+	}
+	qctx.PreemptedResourceByPriorityClass[pcName].Add(rl)
```

Comment:
This line will have no effect, you need to add the value returned by add to something